python/paddle/tensor/manipulation.py

```diff
@@ -1447,7 +1447,7 @@ def expand(x, shape, name=None):
     def get_attr_expand_shape(list_expand_shape):
         attrs_expand_shape = []
         for idx, shape in enumerate(list_expand_shape):
-            if isinstance(shape, Variable):
+            if isinstance(shape, Variable) or shape == -1:
```

Comment:
Should we also consider `shape == None` ? Because we also used `None` to represent unknown shape.

Reply:
if shape == None means dynamic shape, maybe we should add that.
i will confirm that.